scripts/test/Muon/loading_tab/loadwidget_presenter_multiple_file_test.py

```diff
@@ -0,0 +1,143 @@
+        self.obj = QtGui.QWidget()
+
+        self.data = MuonLoadData()
+        self.load_file_view = BrowseFileWidgetView(self.obj)
+        self.load_run_view = LoadRunWidgetView(self.obj)
+        self.load_file_model = BrowseFileWidgetModel(self.data)
+        self.load_run_model = LoadRunWidgetModel(self.data)
+
+        self.view = LoadWidgetView(parent=self.obj, load_file_view=self.load_file_view,
+                                   load_run_view=self.load_run_view)
+        self.presenter = LoadWidgetPresenter(self.view, LoadWidgetModel(self.data))
+        self.presenter.set_load_file_widget(BrowseFileWidgetPresenter(self.load_file_view, self.load_file_model))
+        self.presenter.set_load_run_widget(LoadRunWidgetPresenter(self.load_run_view, self.load_run_model))
+
+        self.view.multiple_loading_check.setCheckState(1)
+        self.presenter.handle_multiple_files_option_changed()
+
+        self.runs = [1234 + i for i in range(10)]
+        self.workspaces = [[run] for run in self.runs]
+        self.filenames = ["C:\dir1\dir2\EMU000" + str(run) + ".nxs" for run in self.runs]
+
+    def tearDown(self):
+        self.obj = None
+
+    def mock_loading_from_browse(self, workspaces, filenames, runs):
+        self.load_file_view.show_file_browser_and_return_selection = mock.Mock(
+            return_value=filenames)
+        self.load_file_model.load_workspace_from_filename = mock.Mock(
+            side_effect=zip(workspaces, runs))
+
+    def mock_loading_from_current_run(self, workspace, filename, run):
+        self.load_run_model.load_workspace_from_filename = mock.Mock(
+            return_value=(workspace, filename, run))
+
+    def mock_user_input_multiple_runs(self, user_input, workspaces, filenames, runs):
+        self.load_run_view.get_run_edit_text = mock.Mock(return_value=user_input)
+        self.load_run_model.load_workspace_from_filename = mock.Mock(
+            side_effect=zip(workspaces, filenames, runs))
+
+    def mock_user_input_multiple_files(self, workspaces, filenames, runs):
+        self.load_file_view.get_file_edit_text = mock.Mock(
+            return_value=";".join(filenames))
+        self.load_file_model.load_workspace_from_filename = mock.Mock(
+            side_effect=zip(workspaces, runs))
+
+    def assert_model_contains_correct_loaded_data(self):
+        # use sorted due to threads finishing at different times
+        self.assertEqual(sorted(self.presenter._model.filenames), sorted(self.filenames))
+        self.assertEqual(sorted(self.presenter._model.workspaces), sorted(self.workspaces))
+        self.assertEqual(sorted(self.presenter._model.runs), sorted(self.runs))
+
+    def assert_interface_contains_correct_runs_and_files(self):
+        self.assertEqual(self.load_file_view.get_file_edit_text(), ";".join(self.filenames))
+        self.assertEqual(self.load_run_view.get_run_edit_text(), "1234-1243")
+
+    def mock_disabling_buttons_in_run_and_file_widget(self):
```

Comment:
minor point, but would this be better above the asserts so all of the mock functions are together?

Reply:
Have moved it up.